x-pack/plugins/snapshot_restore/server/plugin.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+declare module 'kibana/server' {
+  interface RequestHandlerContext {
+    snapshotRestore?: SnapshotRestoreContext;
+  }
+}
+
+import { first } from 'rxjs/operators';
+import { i18n } from '@kbn/i18n';
+import {
+  CoreSetup,
+  Plugin,
+  Logger,
+  PluginInitializerContext,
+  IScopedClusterClient,
+} from 'kibana/server';
+
+import { PLUGIN } from '../common';
+import { License } from './services';
+import { ApiRoutes } from './routes';
+import { isEsError, wrapEsError } from './lib';
+import { elasticsearchJsPlugin } from './client/elasticsearch_sr';
+import { Dependencies } from './types';
+import { SnapshotRestoreConfig } from './config';
+
+export interface SnapshotRestoreContext {
+  client: IScopedClusterClient;
+}
+
+export class SnapshotRestoreServerPlugin implements Plugin<void, void, any, any> {
+  private readonly logger: Logger;
+  private readonly apiRoutes: ApiRoutes;
+  private readonly license: License;
+
+  constructor(private context: PluginInitializerContext) {
+    const { logger } = this.context;
+    this.logger = logger.get();
+    this.apiRoutes = new ApiRoutes();
+    this.license = new License();
+  }
+
+  public async setup(
```

Comment:
Not sure when, but `async` setup is going to be deprecated in the future and it will become sync code only

Reply:
Ok. Let's wait and see. I guess we'll all have to go and update all our plugins then.